app/services/MatcherPool.scala

```diff
@@ -215,6 +215,9 @@ class MatcherPool(
     }
   }
 
+  private def doesMatcherServeAllCategories(categoryIds: List[String], matcher: Matcher) =
```

Comment:
This actually seems to be asking if the matcher serves any category in the provided list, rather than the all in the function name? `canMatcherGenerateResultsForAnyCategory`?

Reply:
The `allCategories` in this name refers to the requested categories, not the matcher categories – that is to say, can the given matcher serve all of the categories in `categoryIds`? There's probably a better name out there – e.g. `doesMatcherServeCategoryIds`?